pipert/core/message.py

```diff
@@ -0,0 +1,107 @@
+from abc import ABC, abstractmethod
+from io import BytesIO
+
+import numpy as np
+import time
+import pickle
+import cv2
+
+from PIL import Image
+
+
+class Payload(ABC):
+
+    def __init__(self, data):
+        self.data = data
+        pass
+
+    @abstractmethod
+    def decode(self):
+        pass
+
+    @abstractmethod
+    def encode(self):
+        pass
+
+
+class FramePayload(Payload):
+
+    def __init__(self, data):
+        super().__init__(data)
+
+    def decode(self):
+        self.data = np.array(Image.open(BytesIO(self.data)))
+
+    def encode(self):
+        _, buf = cv2.imencode(".jpg", self.data)
+        self.data = buf.tobytes()
+
+
+class PredictionPayload(Payload):
+    def __init__(self, data):
+        super().__init__(data)
+
+    def decode(self):
+        pass
+
+    def encode(self):
+        pass
+
+
+class Message:
+    next_msg_id = 0
+
+    def __init__(self, data, stream_address):
+        if isinstance(data, np.ndarray):
+            self.payload = FramePayload(data)
+        else:
+            self.payload = PredictionPayload(data)
+        self.stream_address = stream_address
+        self.history = {}
+        self.id = Message.next_msg_id
+        Message.next_msg_id += 1
```

Comment:
This can cause ambiguity when using multiprocessing, since every process will have it's own Message `next_msg_id`. Maybe add the name of the data_source / component_name to resolve ambiguity?